cluster-http/src/main/scala/akka/cluster/http/ClusterHttpApi.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.cluster.http
+
+import akka.actor.AddressFromURIString
+import akka.cluster.{Cluster, Member}
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.RouteResult
+import akka.stream.ActorMaterializer
+import spray.json._
+
+import scala.concurrent.Future
+
+final case class ClusterUnreachableMember(node: String, observedBy: Seq[String])
+final case class ClusterMember(node: String, nodeUid: String, status: String, roles: Set[String])
+final case class ClusterMembers(selfNode: String, members: Set[ClusterMember], unreachable: Seq[ClusterUnreachableMember])
+final case class ClusterHttpApiMessage(message: String)
+
+object ClusterHttpApiOperation extends Enumeration {
```

Comment:
is there any benefit of Enumeration over having a trait and case objects extending it?